### PR TITLE
Docker buildx bake - The Sequel

### PR DESCRIPTION
**Pull request by back-2-95**
- 2 distros: `bullseye` and `alpine 3.17`
- Main `Dockerfile` + distro specific builder dockerfiles
- Bake definition file which can be extended to more PHP versions and distros
- `.dockerignore` file updated
- Simple buildx bake GHA workflow (no push enabled, just showing) See run: https://github.com/back-2-95/frankenphp/actions/runs/3917789332/jobs/6697797558
- Supersedes https://github.com/dunglas/frankenphp/pull/64

## TODO / TBD

- Add Docker Hub log into GHA workflow and enable push
- Builder dockerfiles can be also combines into one multistage dockerfile

## Notes

- Uses named build contexts introduced in buildx 0.8.0 - No more ARGS

## Testing

Show what is about to go down:

```
docker buildx bake -f docker-bake.hcl --print
```

Build all images locally for arm64 arch (with `--load` and `--set "*.platform=linux/arm64"` flags):

```
docker buildx bake -f docker-bake.hcl --pull --load --set "*.platform=linux/arm64"
```

Same as previous but build only 1 target from `docker-bake.hcl`:

```
docker buildx bake -f docker-bake.hcl --pull --load --set "*.platform=linux/arm64" bullseye
```

Build locally and override image name:

```
IMAGE_NAME=my/frankenphp docker buildx bake -f docker-bake.hcl --pull --load --set "*.platform=linux/arm64" --progress plain
```

Build and push all images with override image name (needs you logged in to Docker Hub or some other registry):

```
IMAGE_NAME=my/frankenphp docker buildx bake -f docker-bake.hcl --pull --push
```

### CODE REVIEW

File: .github/workflows/tests.yaml [hunk 1]
```diff
@@ -1,94 +0,0 @@
...
-          tags: frankenphp:${{ github.sha }}-builder
-          builder: ${{ steps.buildx.outputs.name }}
-          cache-from: type=gha
-          cache-to: type=gha,mode=max
-          outputs: type=docker,dest=/tmp/.builder.tar
-
-      - name: Run tests
-        run: |
-          docker load -i /tmp/.builder.tar
-          docker run --rm frankenphp:${{ github.sha }}-builder "go test -a -v"
```
Comment (dunglas) [inline .github/workflows/tests.yaml:38]: We need to continue running the tests somewhere in the CI.

File: builder-bullseye.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,46 @@
+FROM php-base
+
+COPY --from=golang-base /usr/local/go/bin/go /usr/local/bin/go
+COPY --from=golang-base /usr/local/go /usr/local/go
+
+# This is required to link the frankenPHP binary to the PHP binary
```
Comment (dunglas) [inline builder-bullseye.Dockerfile:6]:
```suggestion
# This is required to link the FrankenPHP binary to the PHP binary
```

(nitpicking)

File: builder-bullseye.Dockerfile [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+        libssl-dev \
+        libxml2-dev \
+        zlib1g-dev \
+        && \
+        apt-get clean
+
+WORKDIR /go/src/app
+
+COPY go.mod go.sum ./
+RUN go mod graph | awk '{if ($1 !~ "@") print $2}' | xargs go get
```
Comment (dunglas) [inline builder-bullseye.Dockerfile:24]: Can you add a comment explaining why it's necessary please?

**Review by dunglas — CHANGES_REQUESTED** [review]

File: .github/workflows/tests.yaml (hunk 1 above)
Comment (back-2-95) [reply .github/workflows/tests.yaml:38]: Yes, this is just that I need help with this one.

File: builder-bullseye.Dockerfile (hunk 3 above)
Comment (back-2-95) [reply builder-bullseye.Dockerfile:24]: Separate file or what?

Comment (dunglas) [reply builder-bullseye.Dockerfile:24]: I don't understand what's it's doing :D

**Comment by dunglas** [comment]
I rebased your branch and fixed and issue with the Dockerfile versions.

**Pull request closed without merging**